en/lessons/libraries/distillery.md

```diff
@@ -0,0 +1,245 @@
+---
+version: 2.0.0
+title: Distillery (Basics)
+---
+
+Distillery
+
+## What is a release?
+
+A release is a package containing your compiled Erlang/Elixir code (i.e [BEAM](https://en.wikipedia.org/wiki/BEAM_(Erlang_virtual_machine)) [bytecode](https://en.wikipedia.org/wiki/Bytecode)). It also provides any scripts necessary for launching your application.
+
+> When you have written one or more applications, you might want to create a complete system with these applications and a subset of the Erlang/OTP applications. This is called a release. - [Erlang documentation](http://erlang.org/doc/design_principles/release_structure.html)
+
+> Releases enable simplified deployment: they are self-contained, and provide everything needed to boot the release; they are easily administered via the provided shell script to open up a remote console, start/stop/restart the release, start in the background, send remote commands, and more. In addition, they are archivable artifacts, meaning you can restore an old release from its tarball at any point in the future (barring incompatibilities with the underlying OS or system libraries). The use of releases is also a prerequisite of performing hot upgrades and downgrades, one of the most powerful features of the Erlang VM. - [Distillery Documentation](https://hexdocs.pm/distillery/introduction/understanding_releases.html)
+
+A release will contain the folowing:
+* a /bin folder
+  * This contains a script that is the starting point to running your entire application
+* a /lib folder
+  * This contains the compiled bytecode of the application along with any dependencies.
+* a /relealses folder
+  * This contains metadata about the release as well as hooks and custom commands
+* a /erts-VERSION
+  * This contains the Erlang runtime which will allow a machine to run your application without having Erland or Elixir installed
+
+
+### Getting started/installation
+
+To add Distillery to your project, add it as a depdency to your `mix.exs` file. *Note* - if you are working on an umbrella app this should be in the mix.exs in the root of your project
+
+```
+defp deps do
+  [{:distillery, "~> 2.0"}]
+end
+```
+
+Then in your terminal call:
+
+```
+mix deps.get
+```
+
+```
+mix compile
+```
+
+
+### Building your release
+
+In your terminal, run
+
+```
+mix release.init
+```
+
+This command generates a rel directory with some configuration files in it.
+
+To generate a release in your terminal run `mix release`
+
+Once the release is built you should see some instructions in your terminal
+
```

Comment:
Might be nice to take a brief look at the out-of-the box configuration and explain some of the highlights? like what comes in the `rel/config.exs` file and what does it do?